be/src/exec/pipeline/hashjoin/spillable_hash_join_probe_operator.cpp

```diff
@@ -0,0 +1,484 @@
+        }
+
+        auto iter = _pid_to_process_id.find(probe_partition->partition_id);
+        if (iter == _pid_to_process_id.end()) {
+            auto mem_table = probe_partition->spill_writer->mem_table();
+            mem_table->append_selective(*chunk, selection.data(), from, size);
+        } else {
+            auto partitioned_chunk = chunk->clone_empty();
+            partitioned_chunk->append_selective(*chunk, selection.data(), from, size);
+            _probers[iter->second]->push_probe_chunk(state, std::move(partitioned_chunk));
+        }
+        probe_partition->num_rows += size;
+    };
+    RETURN_IF_ERROR(_probe_spiller->partitioned_spill(state, chunk, hash_column.get(), partition_processer, exexutor,
+                                                      spill::MemTrackerGuard(tls_mem_tracker)));
+
+    return Status::OK();
+}
+
+Status SpillableHashJoinProbeOperator::_load_partition_build_side(RuntimeState* state,
+                                                                  const std::shared_ptr<spill::SpillerReader>& reader,
+                                                                  size_t idx) {
+    TRY_CATCH_ALLOC_SCOPE_START()
+    SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(state->instance_mem_tracker());
+    auto builder = _builders[idx];
+    bool finish = false;
+    while (!finish && !_is_finished) {
+        if (state->is_cancelled()) {
+            return Status::Cancelled("cancelled");
+        }
+        RETURN_IF_ERROR(
+                reader->trigger_restore(state, spill::SyncTaskExecutor{}, spill::MemTrackerGuard(tls_mem_tracker)));
+        auto chunk_st = reader->restore(state, spill::SyncTaskExecutor{}, spill::MemTrackerGuard(tls_mem_tracker));
+        if (chunk_st.ok() && chunk_st.value() != nullptr && !chunk_st.value()->is_empty()) {
+            RETURN_IF_ERROR(builder->append_chunk(state, std::move(chunk_st.value())));
+        } else if (chunk_st.status().is_end_of_file()) {
+            RETURN_IF_ERROR(builder->build(state));
+            finish = true;
+        } else if (!chunk_st.ok()) {
+            return chunk_st.status();
+        }
+    }
+    if (finish) {
+        DCHECK_EQ(builder->hash_table_row_count(), _processing_partitions[idx]->num_rows);
+    }
+    TRY_CATCH_ALLOC_SCOPE_END()
+    return Status::OK();
+}
+
+Status SpillableHashJoinProbeOperator::_load_all_partition_build_side(RuntimeState* state) {
+    auto spill_readers = _join_builder->spiller()->get_partition_spill_readers(_processing_partitions);
+    _latch.reset(_processing_partitions.size());
+    auto query_ctx = state->query_ctx()->weak_from_this();
+    for (size_t i = 0; i < _processing_partitions.size(); ++i) {
+        std::shared_ptr<spill::SpillerReader> reader = std::move(spill_readers[i]);
+        auto task = [this, state, reader, i, query_ctx]() {
+            if (query_ctx.lock()) {
+                _update_status(_load_partition_build_side(state, reader, i));
```

Comment:
what if some ok, others not?

Reply:
_update_status will make next push_chunk return error